### PR TITLE
Feature/Tenant Token

### PR DESCRIPTION
**Pull request by brunoocasali**
Create the `generateTenantToken` following the specification https://github.com/meilisearch/specifications/pull/89

### CODE REVIEW

File: src/Meilisearch/MeilisearchClient.cs [hunk 1]
```diff
@@ -327,6 +331,20 @@ public async Task<bool> DeleteKeyAsync(string keyUid, CancellationToken cancella
...
 
+        /// <summary>
+        /// Generate a tenant token to be used during search.
+        /// </summary>
+        /// <param name="searchRules">Object with the rules to be inject in the search call.</param>
+        /// <param name="apiKey">API Key which signs the generated token.</param>
+        /// <param name="expiresAt">Date to express how long the generated token will last. If null the token will last forever.</param>
+        /// <exception cref="MeilisearchTenantTokenApiKeyInvalid">When there is no <paramref name="apiKey"/> defined in the client or as argument.</exception>
+        /// <exception cref="MeilisearchTenantTokenExpired">When the sent <paramref name="expiresAt"/> param is in the past</exception>
+        /// <returns>Responds with the generated token.</returns>
```
Comment (alallema) [inline src/Meilisearch/MeilisearchClient.cs:342]:
```suggestion
        /// <returns>Returns a generated tenant token.</returns>
```

**Review by alallema — APPROVED** [review]
I put a suggestion on a comment but as it is not essential so I approve! 🎉

File: src/Meilisearch/MeilisearchClient.cs [hunk 2]
```diff
@@ -327,6 +331,20 @@ public async Task<bool> DeleteKeyAsync(string keyUid, CancellationToken cancella
             return responseMessage.StatusCode == HttpStatusCode.NoContent;
         }
 
+        /// <summary>
+        /// Generate a tenant token to be used during search.
+        /// </summary>
+        /// <param name="searchRules">Object with the rules to be inject in the search call.</param>
```
Comment (bidoubiwa) [inline src/Meilisearch/MeilisearchClient.cs:337]:
```suggestion
        /// <param name="searchRules">Object with the rules applied in a search call.</param>
```

**Review by bidoubiwa — APPROVED** [review]
LGTM 🔥🔥

**Comment by brunoocasali** [comment]
@alallema and @bidoubiwa since GH will dismiss your reviews if I implement your suggestions here, I will submit a new PR :)

**Pull request merged**